### ISSUE TITLE
Please auto-skip when the YouTube title is too far off from song title

### ISSUE THREAD

**Issue by forteller** [KRTirtho/spotube#546]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Pain

Hi. Thanks for a great new version 3.0!

Unfortunately I too often experience that the audio of a seemingly random video is played when I play a playlist. 

![image](https://github.com/KRTirtho/spotube/assets/343561/aa8f297b-ab88-4e7e-91ec-b472e505f8df)
![image](https://github.com/KRTirtho/spotube/assets/343561/3e4c3cbe-214e-412c-874b-6a72dd0dcd0b)

I guess that the correct song is not available at YT at all, and that Spotube just automatically plays whatever video comes up on the top of the search results. This is quite jarring, when you're listening to maybe some soothing music, and then a TED talk suddenly starts playing.

### Suggested solution

I think the best solution is a check to see how different the top results title is from the song title. If it does not all of the words of the song title, or some other kind of check, it should just be automatically skipped.

Other solutions might be found, that I just haven't thought of yet.

### Additional information

Thanks again!

**Comment by RaptaG** [comment]
I suggest you use the Piped API with the `kavin.rock libre (Official)` instance, it has a very high level of accuracy comparing to the YT one.

![image](https://github.com/KRTirtho/spotube/assets/77157639/be1c2e9f-98d3-4889-91e1-4d57b36740ca)

**Comment by forteller** [comment]
Thanks, I tried, but then nothing played. I tried the default instance and two other

**Comment by RaptaG** [comment]
@forteller try the kavin.rocks, kavin.rocks (libre) and whatever.social ones and see which one works. They should all work, but do test it for yourself

**Comment by abhisheksharma886** [comment]
Hello, I would like to work on this, can it be assigned to me

**Comment by KRTirtho** [comment]
That's super nice of you ❤️

**Comment by LetsBuild-exe** [comment]
Yes this is a great feature, cause sometimes when i have some playlist it just randomly starts playing an indian / oriental music...
